x-pack/test_serverless/shared/services/svl_user_manager.ts

```diff
@@ -94,24 +97,48 @@ export function SvlUserManagerProvider({ getService }: FtrProviderContext) {
       return this.createApiKeyForRole(this.DEFAULT_ROLE);
     },
     async createApiKeyForRole(role: string): Promise<RoleCredentials> {
-      const cookieHeader = await this.getApiCredentialsForRole(role);
+      // Get admin credentials in order to create the API key
+      const adminCookieHeader = await this.getApiCredentialsForRole('admin');
+
+      // Get the role descrtiptor for the role
+      let roleDescriptors = {};
+      if (role !== 'admin') {
+        const roleDescriptor = supportedRoleDescriptors[role];
+        if (!roleDescriptor) {
+          throw new Error(`Cannot create API key for non-existent role "${role}"`);
+        }
+        log.debug(
+          `Creating api key for ${role} role with the following privileges ${JSON.stringify(
+            roleDescriptor
+          )}`
+        );
+        roleDescriptors = {
+          [role]: roleDescriptor ?? {
+            cluster: [],
+            index: [],
+            applications: [],
+          },
+        };
+      }
+
+      log.debug(`Creating api key for default role: [${this.DEFAULT_ROLE}]`);
```

Comment:
I got confused by this message. We probably don't need it at all?

```
       │ debg new fake SAML authentication with 'admin' role
       │ debg Creating api key for viewer role with the following privileges {"cluster":["manage_own_api_key","read_pipeline"],"indices":[{"names":["*"],"privileges":["read","view_index_metadata"]}],"applications":[{"application":"kibana-.kibana","privileges":["read"],"resources":["*"]}]}
       │ debg Creating api key for default role: [developer]
```

Reply:
Yeah I just forgot to delete it from the original implementation.